src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -135,7 +135,11 @@ public static Tag parseTag(String tag) throws ParseException {
     public static Set<Tag> parseTags(Collection<String> tags) throws ParseException {
         requireNonNull(tags);
         final Set<Tag> tagSet = new HashSet<>();
+        final Set<String> uniqueTags = new HashSet<>();
         for (String tagName : tags) {
+            if (!uniqueTags.add(tagName)) { // If adding returns false, tagName is a duplicate
+                throw new ParseException(Tag.DUPLICATE_TAG);
```

Comment:
Has this been tested? Maybe we can add a test in ParserUtilTest if its allowed to make sure it behaves correctly?